src/main/scala/io/computenode/cyfra/dsl/GContext.scala

```diff
@@ -46,17 +49,12 @@ object Empty:
   given GStructSchema[Empty] = derived
 
 class MVPContext extends GContext {
-  implicit val ec: ExecutionContextExecutor = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(16))
+  // stub for execute
+  def execute[R](pipeline: ComputePipeline, inData: Seq[ByteBuffer])(using uniformContext: UniformContext[_]) = ???
```

Comment:
As per some comment up ^ I think Context could have GFunction passed to execute.

Reply:
Something like:
```scala
def execute(input: GMem[H], fn: GFunction[_, H, R], output: WritableGMem[R, _]): Future[Unit]
```
Not sure where to get the `G` type param, is it inferred?

Reply:
Something like
```scala
def execute[G <: GStruct[G] : Tag: GStructSchema, H <: Value: Tag: FromExpr, R <: Value: Tag: FromExpr](pipeline: ComputePipeline, inData: Seq[ByteBuffer], fn: GFunction[_, _, R])(using uniformContext: UniformContext[_]) = ???
```
?

(I'll have to change it in `GContext` as well.)

Reply:
See above, I'll push the attempt I made so far.